[SofaGeneralObjectInteraction] Fix AttachConstraint in case of FreeMotion (LM solving)

tgaugry opened the pull request:
- Fix Attachconstraint for use with FreeMotionAnimationLoop - credit to @hugtalbot #940
- refactor of parameters to use `SOFA_UNUSED`
- fix dynamicConstraintFactor



______________________________________________________
<!--- Please leave this at the end of your message -->
This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

Commit:
Fix attachconstraint while using FreeMotionAnimationLoop + make use of m_dynamicFactor

damienmarchal commented on modules/SofaGeneralObjectInteraction/AttachConstraint.h:
```diff
@@ -163,11 +166,9 @@ protected :
-            Real constraintFactor = getConstraintFactor(index);
-            dx1 += in2*(d_responseFactor.getValue()*constraintFactor);
-            dx2 += in1*(d_responseFactor.getValue()*constraintFactor);
+            Deriv corr = (dx2-dx1)*0.5*d_responseFactor.getValue()*getConstraintFactor(index);
```
Try to keep the data_field out of loops. Here the projectResponse function is called n-time...wile d_responseFactor is "assumed" constant during the time step. So pass the response factor as an argument.

damienmarchal commented on modules/SofaGeneralObjectInteraction/AttachConstraint.inl:
```diff
@@ -258,7 +258,7 @@ void AttachConstraint<DataTypes>::init()
     this->core::behavior::PairInteractionProjectiveConstraintSet<DataTypes>::init();
 
-    dynamicConstraintFactor = !d_constraintFactor.isSet();
+    m_dynamicConstraintFactor = !d_constraintFactor.isSet();
```
Shouldn't this thing go in reinit ?

Commit:
Remove dynamicConstraintFactor + use ReadAccessor + move factor to parent function

Commit:
refactor/prettifying

damienmarchal committed:
[SofaGeneralObjectInteraction] Remove component_state in AttachConstraint

In the current state there is no need for ComponentState as the "error" message
are in fact "warning/info" because the detected problems are not preventing the
object to function properly.

I did various minor cleaning (to remove compiler warnings).
There is still a lot of uglyness in this class if a brave hacker found it,
don't hesitate.

tgaugry commented on modules/SofaGeneralObjectInteraction/AttachConstraint.inl:
```diff
@@ -268,17 +240,17 @@ void AttachConstraint<DataTypes>::reinit()
         if(constraintFactor.size() != f_indices2.getValue().size())
         {
-            msg_error() << "Size of vector constraintFactor, do not fit number of indices attached (" << constraintFactor.size() << " != " << f_indices2.getValue().size() << ").";
+            msg_warning() << "Size of vector constraintFactor, do not fit number of indices attached (" << constraintFactor.size() << " != " << f_indices2.getValue().size() << ").";
```
@damienmarchal Not having an error can lead to crash if the the number of indices isn't the same, do we really want that ?

tgaugry committed:
Merge branch 'master' into fix-attachconstrain-fremotionanimationloop

tgaugry committed:
Update AttachConstraint.h
